wpimath/src/main/java/edu/wpi/first/math/estimator/DifferentialDrivePoseEstimator.java

```diff
@@ -255,13 +256,12 @@ public Pose2d getEstimatedPosition() {
    *     source in this case.
    */
   public void addVisionMeasurement(Pose2d visionRobotPoseMeters, double timestampSeconds) {
-    m_latencyCompensator.applyPastGlobalMeasurement(
-        Nat.N3(),
-        m_observer,
-        m_nominalDt,
-        StateSpaceUtil.poseTo3dVector(visionRobotPoseMeters),
-        m_visionCorrect,
-        timestampSeconds);
+    m_visionCorrect.accept(
+        new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.0, 0.0, 0.0),
+        StateSpaceUtil.poseTo3dVector(
+            getEstimatedPosition()
+                .transformBy(
+                    visionRobotPoseMeters.minus(m_poseBuffer.getSample(timestampSeconds)))));
```

Comment:
m_poseBuffer.getSample(timestampSeconds) can return null, but this isn't checked.